### PR TITLE
Fixes for open tests

### PR DESCRIPTION
**Pull request by mikbras**
Add fixes so that the following ltp open tests will pass.

```
/ltp/testcases/kernel/syscalls/open/open03
/ltp/testcases/kernel/syscalls/open/open07
/ltp/testcases/kernel/syscalls/open/open09
```

This involved the following changes:
- access checks in read and write functions (O_RDONLY/O_WRONLY)
- add umask syscall
- add chown syscall as no-op
- add setpgid syscall
- update ltp tests to use ext2 (instead of ramfs)
- fix handling of filenames ending in slash (denoting /.)

### CODE REVIEW

File: include/myst/process.h [hunk 1]
```diff
@@ -8,6 +8,8 @@
 #include <sys/types.h>
 #include <unistd.h>
 
+#define MYST_DEFAULT_UMASK (S_IWGRP | S_IWOTH)
```
Comment (jxyang) [inline include/myst/process.h:11]: Why do we need permission `S_IWOTH`?

File: kernel/umask.c [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#include <myst/syscall.h>
+#include <myst/spinlock.h>
+#include <myst/thread.h>
+#include <myst/eraise.h>
+
+long myst_syscall_umask(mode_t mask)
```
Comment (jxyang) [inline kernel/umask.c:6]: Don't feel we should dedicate a file just for a single function. The cons are increased compile time and decreased code readability.

Maybe move this and a few other file permission related functions to a separate file?

File: kernel/syscall.c [hunk 3]
```diff
@@ -3074,13 +3074,28 @@ long myst_syscall(long n, long params[6])
             BREAK(_return(n, myst_syscall_fchmod(fd, mode)));
         }
         case SYS_chown:
-            break;
+        {
+            const char* pathname = (const char*)x1;
+            uid_t owner = (uid_t)x2;
+            gid_t group = (gid_t)x3;
```
Comment (jxyang) [inline kernel/syscall.c:3080]: Is this needed for this PR?

File: kernel/syscall.c [hunk 4]
```diff
@@ -3181,7 +3196,18 @@ long myst_syscall(long n, long params[6])
             BREAK(_return(n, MYST_DEFAULT_GID));
         }
         case SYS_setpgid:
-            break;
+        {
+            gid_t gid = (gid_t)x1;
```
Comment (jxyang) [inline kernel/syscall.c:3200]: Is this needed for this PR?

File: kernel/syscall.c (hunk 3 above)
Comment (jxyang) [reply kernel/syscall.c:3080]: Some other ltp tests require SYS_chown. Let's keep this.

File: include/myst/process.h (hunk 1 above)
Comment (mikbras) [reply include/myst/process.h:11]: This is is the "typical default value" as defined by the umask manpage.

```
 typical default value for the process  umask  is  S_IWGRP | S_IWOTH
       (octal  022).
```

File: kernel/umask.c (hunk 2 above)
Comment (mikbras) [reply kernel/umask.c:6]: Fixed. I moved it to kernel/fs.c (since it is file-system related).

File: kernel/syscall.c (hunk 4 above)
Comment (mikbras) [reply kernel/syscall.c:3200]: Yes, one of the LTP tests called it.

**Comment by hukoyu** [comment]
@mikbras I need this PR to be merged so that I can submit PR to enable more LTP tests. Most of the LTP tests call chown syscall.

**Review by jxyang — APPROVED** [review]

**Pull request merged**